rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala

```diff
@@ -29,6 +29,14 @@ package object accounting extends Costs {
       override def set(s: Cost): F[Unit]            = state.set(s)
     }
 
+  def noCharge[F[_]: Monad, A](fa: F[A])(implicit cost: _cost[F]): F[A] =
```

Comment:
Wouldn't it make more sense to instantiate a one-off `_cost` instance here instead?

Reply:
I fear that the code below isn't thread-safe

Reply:
That's a very interesting idea! This would also cause the `charge`-s conducted in those blocks to not show up in the upstream cost log. Which is desirable, I think.

Reply:
We'd have to make sure though that the newly-spawned _cost instance is being used actaully. Functions have no implicit parameters, so I guess we'd have an explicit one and define an `implicit val` in the function block?

Reply:
Another idea that comes to my mind would be to instantiate here a `noOpCost` which would never go below zero and thus never raise errors

Reply:
No matter what we spawn, I'd do that inside the `def noCharge`, as it seems to clean-up some boilerplate. See my other comments :)

Reply:
I'd also say a throw-away local-to-noCharge CostAccount.of(Cost.Max) is good enough :+1:

Reply:
Semantically, they're equivalent, but I think we should aim for a unified `_cost` instance. This makes future changes less finicky because we won't have to worry about synchronizing `_cost` instances. It has the added benefit of using less memory. It uses one instance of `_cost` rather than 1 + (# of `noCharge` calls) instances.